node/node.go

```diff
@@ -56,6 +60,7 @@ func New(
 	blockdb,
 	mempooldb dbm.DB,
 	txdb cometdb.DB,
+	ethstatedb *state.StateDB,
```

Comment:
Can this be an interface instead? (apologies if we discussed this on another thread and I forgot)

Reply:
Like this https://github.com/ethereum/go-ethereum/blob/ac0f220040bd4271a040170bc0b039eb88dee9bd/core/vm/interface.go#L32

Reply:
Nvm, just saw your comment in the other github thread about obtaining the trie and responded to it. I like this `*state.StateDB` now :)